xc/device.ml

```diff
@@ -1143,18 +1143,67 @@ module PCI = struct
     (* Sort into the order the devices were plugged *)
     List.sort (fun a b -> compare (fst a) (fst b)) pairs
 
-  let add ~xs pcidevs domid =
+  let _pci_add ~xc ~xs pcidev domid slot =
+    let sysfs_pci_dev = "/sys/bus/pci/devices/" in
+    let bus = 0 in
+    let func = 0 in
+    let pci_bdf = Printf.sprintf "%04x:%02x:%02x.%01x" domid bus slot func in
```

Comment:
This is a PCI SBDF, not BDF, and the guests domid is definitely not what you want in the top.

No x86 hardware currently exists with a Segment different to 0, but we will get there eventually.  Qemu is definitely not capable of dealing with a nonzero segment at the moment.

Reply:
We have a type for this already: `Xenops_interface.Pci.address`, and helper function `Xenops_interface.Pci.string_of_address`. Code is in xcp-idl.git/lib/xcp_pci.ml.

Reply:
Note that what is called "domain" there is ambiguous ;) It's not Xen domain.

Reply:
That is why they are more commonly referred to as Segment rather then Domain.

Reply:
I've updated the code with a record type for the 'bdf' info, I can also change the var name if you wish. Let me know if this is fixes your comments

Reply:
The code was updated to use Pci.address type, and set "domain" i.e. segment to 0:
```